File: LeetCode/347-top-k-frequent-elements/347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer,Integer> numofRepeate=new HashMap<>();
        List<Integer>[] bucketSort=new ArrayList[nums.length+1];
        int[] result=new int[k];
        for(int i=0;i<nums.length;i++){
            numofRepeate.merge(nums[i],1,(oldVal,newVal)->oldVal+newVal);
        }
        
        numofRepeate.forEach((key,value)->{
            if(bucketSort[value]!=null)
                bucketSort[value].add(key);
            else{
                List<Integer> l=new ArrayList<>();
                l.add(key);
                bucketSort[value]=l;
            }        
        });
        int pt=0;
        for(int i=bucketSort.length-1;i>=0 && pt<k;i--){
            if(bucketSort[i]!=null){
                List<Integer> currentList=bucketSort[i];
                for(int j=0;j<currentList.size();j++){
                    result[pt++]=currentList.get(j);
                }
            }
        }        
        return result;
    }
}

File: LeetCode/347-top-k-frequent-elements/347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freqMap = {}
        freqlist = [[] for i in range(len(nums) + 1)]
        result = []
        for val in nums:
            freqMap[val] = freqMap.setdefault(val, 0) + 1
        for key, value in freqMap.items():
            freqlist[value].append(key)
        for i in range(len(freqlist) - 1, 0, -1):
            for val in freqlist[i]:
                result.append(val)
                if len(result) == k:
                    return result
        return result